### PR TITLE
Document the complete artifact endpoint

### PR DESCRIPTION
**Pull request by jhford**

### CODE REVIEW

**Comment by jhford** [comment]
Please merge if you're OK with the comment

File: src/artifacts.js [hunk 1]
```diff
@@ -660,7 +660,17 @@ api.declare({
...
   title:      'Complete Artifact',
-  description: 'tbd',
+  description: [
+    'This endpoint finalises an upload done through the blob `storageType`.',
+    'The queue will ensure that the task/run is still allowing artifacts',
+    'to be uploaded.  For single-part S3 blob artifacts, this endpoint',
+    'will simply ensure the artifact is present in S3.  For multipart S3',
+    'artifacts, the endpoint will perform the commit step of the multipart',
+    'upload flow.  As the final step for both multi and single part artifacts,',
+    'the `present` entity field will be set to `true` to reflect that the',
```
Comment (jonasfj) [inline src/artifacts.js]: Present field is internal...

This publishes the artifact completed message..

File: src/artifacts.js [hunk 2]
```diff
@@ -660,7 +660,17 @@ api.declare({
   deferAuth:  true,
   input:      'put-artifact-request.json#',
   title:      'Complete Artifact',
-  description: 'tbd',
+  description: [
+    'This endpoint finalises an upload done through the blob `storageType`.',
+    'The queue will ensure that the task/run is still allowing artifacts',
+    'to be uploaded.  For single-part S3 blob artifacts, this endpoint',
```
Comment (jonasfj) [inline src/artifacts.js]: Consumers shouldn't need to know about queue internals like this...

File: src/artifacts.js [hunk 3]
```diff
@@ -660,7 +660,17 @@ api.declare({
...
+    'will simply ensure the artifact is present in S3.  For multipart S3',
```
Comment (jonasfj) [inline src/artifacts.js]: Empty lines to split paragraphs would be nice to have :)

**Pull request merged**